physics/discrete_trajectory.hpp

```diff
@@ -256,11 +259,11 @@ class DiscreteTrajectory : public Forkable<DiscreteTrajectory<Frame>,
       std::vector<DiscreteTrajectory<Frame>**> const& forks);
 
   // Returns the Hermite interpolation for the left-open, right-closed
-  // trajectory segment containing the given |time|, or, if |time| is |t_min()|,
+  // trajectory segment bounded above by |upper|, or, if |upper| is |begin()|,
```

Comment:
I think that the special case `upper == begin` is not longer used, right?  If so, I'd remove that part of the comment and leave the contract "fuzzy".